src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+     */
+    onOpen?: () => void;
+    /**
+     * Optional: Define a callback for when Popover content is closed
+     */
+    onClose?: () => void;
+}
+
+export const Popover: React.FC<PopoverProps> = ({
+    children,
+    content = '',
+    placement = 'bottom-start',
+    offset = 5,
+    isOpen = false,
+    onOpen,
+    onClose
+}: PopoverProps) => {
+    const [triggerReference, setTriggerReference] = React.useState(undefined);
+    const [popperReference, setPopperReference] = React.useState(undefined);
+    const popoverTriggerRef = React.useRef<HTMLDivElement>(null);
+    const popoverContentRef = React.useRef<HTMLDivElement>(null);
+
+    const [openByDefault, setOpenByDefault] = React.useState(isOpen);
+
+    const [render, setRender] = React.useState(openByDefault);
+
+    const { styles, attributes } = usePopper(triggerReference, popperReference, {
+        placement,
+        strategy: 'fixed',
+        modifiers: [
+            {
+                name: 'offset',
+                enabled: !!offset,
+                options: {
+                    offset: [0, offset]
+                }
+            },
+            {
+                name: 'flip',
+                enabled: true
+            }
+        ]
+    });
+
+    const resolveCallback = React.useCallback(
+        state => {
+            if (onClose && !state) onClose();
+            if (onOpen && state) onOpen();
+        },
+        [onClose, onOpen]
+    );
+
+    const hidePopover: () => void = React.useCallback(() => {
+        if (openByDefault) {
+            setOpenByDefault(false);
+        } else {
+            setRender(false);
+        }
+        resolveCallback(false);
+    }, [openByDefault, resolveCallback]);
+
+    const handleClose = React.useCallback(() => {
+        if (render) {
+            hidePopover();
+        }
+    }, [render, hidePopover]);
+
+    const handleClick: () => void = React.useCallback(() => {
+        if (render) {
+            hidePopover();
+        } else {
+            setRender(true);
+            resolveCallback(true);
+        }
+    }, [resolveCallback, setRender, render, hidePopover]);
+
+    // handleOut - handles click outside the trigger
+    const handleOut = React.useCallback(
+        ev => {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            if (popoverTriggerRef && popoverTriggerRef.current && !popoverTriggerRef.current.contains(ev.target)) {
+                if (!openByDefault) {
+                    handleClose();
+                } else {
+                    setOpenByDefault(false);
+                    resolveCallback(false);
+                }
+            }
+        },
+        [openByDefault, popoverTriggerRef, handleClose, resolveCallback]
+    );
+
+    const handleKeyControl = (ev: React.KeyboardEvent<HTMLElement>) => {
+        // eslint-disable-next-line default-case
+        switch (ev.keyCode) {
```

Comment:
What advantage does this switch bring over regular if else? I find it less semantic and we are not using the default case, so for me it looks unnecessary